File: baseutils/src/main/java/com/ec/www/view/swiperecycler/BaseViewHolder.java
package com.ec.www.view.swiperecycler;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;

/**
 * Created by huang on 2017/5/11.
 */

public class BaseViewHolder<T extends ViewDataBinding> extends RecyclerView.ViewHolder {

    public final T bind;

    public BaseViewHolder(T bind) {
        super(bind.getRoot());
        this.bind = bind;
    }

}

File: baseutils/src/main/java/com/ec/www/view/swiperecycler/base/BaseViewHolder.kt
package com.ec.www.view.swiperecycler.base

import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView

/**
 * Created by huang on 2017/5/11.
 */

class BaseViewHolder<T : ViewDataBinding>(val bind: T) : RecyclerView.ViewHolder(bind.root)
